cocos/base/CCScheduler.cpp

```diff
@@ -464,26 +464,27 @@ void Scheduler::schedulePerFrame(const ccSchedulerFunc& callback, void *target,
     HASH_FIND_PTR(_hashForUpdates, &target, hashElement);
     if (hashElement)
     {
+        CCAssert(!hashElement->entry->markedForDeletion, "All elements in _hashForUpdate must not be marked for deletion.");
```

Comment:
Add assert can make developers notice the error, but it doesn't fix the issue. I think we should find a way to fix it.